pkg/cloudprovider/providers/aws/aws.go

```diff
@@ -1490,6 +1490,26 @@ func (c *Cloud) getAwsInstance(nodeName types.NodeName) (*awsInstance, error) {
 	return awsInstance, nil
 }
 
+// wrapAttachError wraps the error returned by an AttachVolume request with
+// additional information, if needed and possible.
+func wrapAttachError(err error, disk *awsDisk, instance string) error {
+	if awsError, ok := err.(awserr.Error); ok {
+		if awsError.Code() == "VolumeInUse" {
+			info, err := disk.describeVolume()
+			if err == nil {
+				for _, a := range info.Attachments {
+					if a.VolumeId == nil || a.State == nil || a.InstanceId == nil {
+						continue
+					} else if *disk.awsID.awsString() == *a.VolumeId && *a.State == "attached" {
```

Comment:
I believe `*disk.awsID.awsString() ` is equivalent to `string(disk.awsID)`.  And it would likely be clearer to do the cast on the other side i.e. `disk.awsID == awsVolumeID(a.VolumeId)`